### PR TITLE
Storages: Separate ColumnFileTinyReader & ColumnFileSetInputStream

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### What is changed and how it works?

1. move `ColumnFileTinyReader` and `ColumnFileSetInputStream` into independent files since we will add `ColumnFileTinyWithVectorIndexReader` and `ColumnFileSetWithVectorIndexInputStream` later.
2. remove `ColumnFileEmptyReader` which is not necessary, so that we can simplify the `ColumnFileSetInputStream`.

```commit-message
Stroages: Separate ColumnFileTinyReader & ColumnFileSetInputStream
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetReader.cpp [hunk 1]
```diff
@@ -116,8 +117,7 @@ ColumnFileSetReaderPtr ColumnFileSetReader::createNewReader(const ColumnDefinesP
 
     for (auto & fr : column_file_readers)
     {
-        new_reader->column_file_readers.push_back(fr->createNewReader(new_col_defs));
-        new_reader->column_file_readers.back()->setReadTag(read_tag);
+        new_reader->column_file_readers.emplace_back(fr ? fr->createNewReader(new_col_defs, read_tag) : nullptr);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetReader.cpp:120]: Is it valid that `column_file_readers` contains a `nullptr`?

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetInputStream.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        }
+        auto skipped_rows = (*cur_column_file_reader)->skipNextBlock();
+        read_rows += skipped_rows;
+        if (skipped_rows)
+            return skipped_rows;
+        else
+        {
+            auto prev = *cur_column_file_reader;
+            ++cur_column_file_reader;
+            prev.reset();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetInputStream.cpp:37]:
```suggestion
            (*cur_column_file_reader).reset();
            ++cur_column_file_reader;
```

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetInputStream.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            block.setStartOffset(read_rows);
+            read_rows += block.rows();
+            return block;
+        }
+        else
+        {
+            auto prev = *cur_column_file_reader;
+            ++cur_column_file_reader;
+            prev.reset();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetInputStream.cpp:63]:
```suggestion
            (*cur_column_file_reader).reset();
            ++cur_column_file_reader;
```

**Review by JaySon-Huang — COMMENTED** [review]
These lines can be simplified

File: dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetReader.cpp (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSetReader.cpp:120]: Get

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**